index.js

```diff
@@ -1850,12 +1850,6 @@ export async function createNodejsBom(path, options) {
           pkgList = pkgList.concat(dlist);
         }
       }
-      return buildBomNSData(options, pkgList, "npm", {
```

Comment:
Should have included a comment here. I think the early return is required to only collect information from package.json alone. This is because for containers we are interested in the post-build lifecycle whereas the lock file could represent the pre-build or build lifecycle and, therefore, might include dev dependencies that may not eventually get included in the image.

As far as the dependency tree goes, I think we can come up with a new function that constructs a tree based on the lock file but only for the components from the package.json files. Such a tree might be useful but not accurate since the dependency tree after tree-shaking could look completely different to build.

Reply:
Sorry for the delay, I'll take a look. Is this tree construction currently done in the same function? in the process of calling `getAllFiles` and then processing the `dependencies` variable?